pkg/virt-controller/watch/vm_initializer.go

```diff
@@ -236,66 +236,77 @@ func checkPresetMergeConflicts(presetSpec *kubev1.DomainSpec, vmSpec *kubev1.Dom
 	return nil
 }
 
-func mergeDomainSpec(presetSpec *kubev1.DomainSpec, vmSpec *kubev1.DomainSpec) error {
-	err := checkPresetMergeConflicts(presetSpec, vmSpec)
-	if err != nil {
-		return err
-	}
+func mergeDomainSpec(presetSpec *kubev1.DomainSpec, vmSpec *kubev1.DomainSpec) (bool, error) {
+	presetConflicts := checkPresetMergeConflicts(presetSpec, vmSpec)
+	applied := false
+
 	if len(presetSpec.Resources.Requests) > 0 {
 		if vmSpec.Resources.Requests == nil {
 			vmSpec.Resources.Requests = k8sv1.ResourceList{}
-		}
-		for key, val := range presetSpec.Resources.Requests {
-			vmSpec.Resources.Requests[key] = val
+			for key, val := range presetSpec.Resources.Requests {
+				vmSpec.Resources.Requests[key] = val
+			}
+			applied = true
 		}
 	}
 	if presetSpec.CPU != nil {
 		if vmSpec.CPU == nil {
 			vmSpec.CPU = &kubev1.CPU{}
+			presetSpec.CPU.DeepCopyInto(vmSpec.CPU)
```

Comment:
Same here - what parts do we update.

Fine for now.

Reply:
At this time, `CPU` is a struct with one entry: `Cores`, do a deep copy is safe and correct here. That could always change in the future of course. :)